### ISSUE TITLE
Master ticket, list of tasks for genome resubmission

### ISSUE THREAD

**Issue by ValWood** [pombase/curation#3345]
Essential
https://github.com/pombase/pombase-chado/issues/1019
~https://github.com/pombase/curation/issues/3344~
~https://github.com/pombase/curation/issues/3337~

Desirable
~https://github.com/pombase/curation/issues/3388~
~https://github.com/pombase/curation/issues/3349~
https://github.com/pombase/curation/issues/3350
~https://github.com/pombase/curation/issues/3351~
https://github.com/pombase/curation/issues/3333

I will extend shortly with the  steps that  I know need to be included.

**Comment by ValWood** [comment]
1. before starting remove
- [ ] /EC_number= (these are legacy)
- [ ] /gene  EMBL uses gene but we haven't maintained , some legacy ones remain. We need to strip them all out.

2. The  features we need to submit are
CDS
3'UTR
5'UTR
LTR
Gap
CDS
tRNA
rRNA
ncRNA
   /ncRNA_class="miRNA"
   http://www.insdc.org/documents/ncrna-vocabulary
 (note that lncRNA, sncRNA, snoRNA, snRNA will ned to be mapped to the ncRNA key/qualifier format)


We don't need to export feature keys
repeat_region
intron
promoter

We need to NOT export the following qualifiers:
/colour
/controlled_curation
/note

We need to export ALL
/db_xref
/product="text"

The following labels need to be remapped:
/systematic_id="SPAC13A11.04c" need to be exported as  /locus_tag="text" (single token)
/primary_name needs to be exported as /standard_name="text"
/synonym=" needs to be exported as /gene_synonym="text"

We need to include the 
/protein_id="<identifier>" from the existing entries.

**Comment by ValWood** [comment]
The proteinID will be tricky. I remember they required this last time, but this isn't real;ly up[ to us so we will try without.

We need to add
FT                   /db_xref="PomBase:SPAC24B11.10c"

and, if we include 
FT                   /db_xref="UniProtKB/Swiss-Prot:Q09897"
ENA should be able to link up to the UniProt protein ID

**Comment by ValWood** [comment]
@kimrutherford  I have assigned this one to you. I was going to try to do it with Artemis EMBL export, but it doesn't really work properly because the ncRNA qualifiers are not recognized work.
I wonder if you could begin to chug through this, ideally we would have a way to create the EMBL format in the outputs so that we can submit them regularly. This will keep our  protein sequence features in line with UniProt and PRO, and  will ensure that resources like AlphaFold  and InterPro are using the correct protein sequence.

I have got an account and  will try to get the validation tool running.

**Comment by kimrutherford** [comment]
> I wonder if you could begin to chug through this,

Will do.

> The features we need to submit are
>   - Gap

EMBL also have "assembly_gap".  Which is best?

**Comment by kimrutherford** [comment]
> LTR

Note to self: this will probably need to be a feature of type `repeat_region` with qualifier `/rpt_type="long_terminal_repeat"`

**Comment by kimrutherford** [comment]
> /systematic_id="SPAC13A11.04c" need to be exported as /locus_tag="text" (single token)

What do you mean by "single token"?

**Comment by kimrutherford** [comment]
> (note that lncRNA, sncRNA, snoRNA, snRNA will ned to be mapped to the ncRNA key/qualifier format)

I can't find any sncRNA features.  Is that a typo?

**Comment by kimrutherford** [comment]
Hi Val.  Sorry for all the comments.  Please ignore until Monday!  I'm about to stop work for the day.

After going through [your comment](https://github.com/pombase/curation/issues/3345#issuecomment-1450636343) and running a test script, there are some qualifiers that you don't mention.  Could you let me know which of these should be included in the EMBL file?  Thanks.

  - [ ] codon_start
  - [ ] estimated_length  (must be an integer in EMBL format, ours are all `/estimated_length=unknown`)
  - [ ] obsolete_name
  - [ ] partial
  - [ ] protein_id
  - [ ] pseudo
  - [ ] shared_id
  - [ ] transl_table

I'm already filtering these two qualifiers:

  - feature_source
  - GO

**Comment by ValWood** [comment]
1. I ticked the ones above that should be included
2. I don't know what shared_id  is ? (cant see it in the docs)
3. Exclude obsolete_name hopefully nobody needs these via EMBL
4. feature source may be mandatory, not sure
5. GO, we should not have any GO qualifiers in the file ?  need fixing? ( ah maybe it is alternative transcripts but I think I can do those in the legacy filoe now?, oerhaps not)
6. sncRNA we use internally, we might need to map these to ncRNA , there is a SO ID if this can be used SO:0002247
7 "single token" I just copied, I think it means cardinality 1 (which is OK as this is our systematic ID)
8. gap (gap in a sequnce)  vs assembly_gap (gap between two components of a genome or transcriptome assembly); they look the the same so either, maybe assembly_gap BONKERS

**Comment by kimrutherford** [comment]
> I realise you men in our data, we definitely have them

There are none in the contig files?

**Comment by kimrutherford** [comment]
> GO, we should not have any GO qualifiers in the file ? need fixing? ( ah maybe it is alternative transcripts but I think I can do those in the legacy filoe now?, oerhaps not)

**Comment by kimrutherford** [comment]
> GO, we should not have any GO qualifiers in the file ? need fixing? ( ah maybe it is alternative transcripts but I think I can do those in the legacy filoe now?, oerhaps not)

That's still a work in progress so there are a few /GO qualifiers.  I'll filter them for EMBL.

**Comment by ValWood** [comment]
How weird?
[SPNCRNA.273](https://www.pombase.org/gene/SPNCRNA.273)
is now an lncRNA. But I definitely changed them (I thought), or are these classified based on their length somehow in Chado?

**Comment by ValWood** [comment]
Actually they are lncRNA (which is a bit weird!) with /SO="SO:0002247"

**Comment by kimrutherford** [comment]
> Actually they are lncRNA (which is a bit weird!) with /SO="SO:0002247"

Should those be changed to ncRNA?

**Comment by kimrutherford** [comment]
I've added my work in progress to SVN here: `pombe-embl/embl_resubmission/*.embl`
I'll update those files as I fix things.

**Comment by kimrutherford** [comment]
I noticed that the [chromosome1 in EMBL - CU329670](https://www.ebi.ac.uk/ena/browser/view/CU329670) has a truncated location `complement(<1..5662)` but we don't have that in our contig file.  Maybe we need to add it?

```
FT   CDS             complement(<1..5662)
FT                   /transl_table=1
FT                   /gene="tlh1"
FT                   /locus_tag="SPAC212.11"
...
```

**Comment by kimrutherford** [comment]
> I don't know what shared_id is ? (cant see it in the docs)

I don't know either.  We have only one occurrence:

```
FT   CDS             join(3802551..3804587,3804661..3804954)
FT                   /colour=12
FT                   /synonym="meu1-1"
FT                   /synonym="SPAC1556.06a"
FT                   /synonym="meu1-2"
FT                   /synonym="SPAC1556.06b"
FT                   /synonym="SPAC1556.06.1"
FT                   /systematic_id="SPAC1556.06"
FT                   /product="Schizosaccharomyces specific protein"
FT                   /shared_id="SPAC1556.06"
...
```

**Comment by kimrutherford** [comment]
I found a very old email that mentions shared_id.  It sounds like it was a GeneDB solution to the multi-transcript gene problem?

The gene where we have our only occurrence is SPAC1556.06, which doesn't have multiple transcripts anymore so we can definitely ignore /shared_id.

-----------------------------------------------

From Val Wood val@sanger.ac.uk via hermes.cam.ac.uk

OK, this is the example I was telling you about yesterday which was the
only one which I ever tried to curate.

It works.... and it doesn't.

The  GeneDB problem is that it shows these as separate gene pages,
because they are curated as separate genes.

They are named as follows
/primary_name="meu1"
/product="sequence orphan"
/synonym="SPAC1556.06"
/systematic_id="SPAC1556.06.1"
/shared_id="SPAC1556.06"

and
/other_transcript="SPAC1556.06.1"
/primary_name="meu2"
/shared_id="SPAC1556.06"
/synonym="SPAC1556.06b"
/synonym="SPAC1556.06"
/systematic_id="SPAC1556.06.2"

In my mind /systematic_id="SPAC1556.06.1" should be the transcript ID
and the systematic ID should be /shared_id="SPAC1556.06"
...I guess this doesn't matter if it all works correctly behind the scenes?

Ensembl will (presumably) handle this as a single gene with different
transcripts?

I think we will ditch the name meu2 and may people use
meu1.1 and meu1.2 for transcript names to match the ID system

Let me know what you think...

**Comment by ValWood** [comment]
>>lncRNA (which is a bit weird!) with /SO="SO:0002247"
>Should those be changed to ncRNA?

Yes, I think so

**Comment by ValWood** [comment]
>I noticed that the [chromosome1 in EMBL - CU329670](https://www.ebi.ac.uk/ena/browser/view/CU329670) has a truncated location complement(<1..5662) but we don't have that in our contig file. Maybe we need to add it?

weird, I always though we did all of those. I revised, and add the /partial qualifier which is necessary with partial coordinates IIRC

**Comment by ValWood** [comment]
shared_id was unnecessary, I deleted it

**Comment by kimrutherford** [comment]
>>lncRNA (which is a bit weird!) with /SO="SO:0002247"
>
>Should those be changed to ncRNA?

Or should they be changed to "sncRNA" since that's what that SO term is?

**Comment by ValWood** [comment]
Let's wait to see if sncRNA survives:
https://github.com/The-Sequence-Ontology/SO-Ontologies/issues/611

It's arbitrary, but it is a useful distinction. I'm inclined not to make any further changes until a better solution is in place.

**Comment by ValWood** [comment]
Could we also remove the (predicted) string at the end of the product lines ? (you  already filter this for chado, but it is still in the contigs...

**Comment by kimrutherford** [comment]
> Let's wait to see if sncRNA survives:

OK, for now all of the RNA features with `/SO="SO:0002247"` (sncRNA) are written as ncRNA features in the EMBL files.

But should we change lncRNA to ncRNA in our contig files if there is a `/SO="SO:0002247"` qualifier?

> Could we also remove the (predicted) string at the end of the product lines 

OK, I've done that.

**Comment by kimrutherford** [comment]
Next step:
> We need to include the
> /protein_id="" from the existing entries.

**Comment by kimrutherford** [comment]
> Next step:
>
>> We need to include the
>> /protein_id="" from the existing entries.

We can get these IDs from EMBL or GenBank for including in the re-submission.

But I'm wondering if we should add them to Chado?  We could then link to the NCBI protein pages: https://www.ncbi.nlm.nih.gov/protein/CAB91572.1/

**Comment by kimrutherford** [comment]
> We can get these IDs from EMBL or GenBank for including in the re-submission.

I've changed the script that generates the files for submission.  It now parses the existing entries from EMBL to extract the /protein_id qualifiers for including in our submission files.

The entries downloaded from EMBL are here:

```
  pombe-embl/embl_resubmission/CU329670.1_chromosome1.embl
  pombe-embl/embl_resubmission/CU329671.1_chromosome2.embl
  pombe-embl/embl_resubmission/CU329672.1_chromosome3.embl
```

**Comment by ValWood** [comment]
I think adding them to Chado would be a good plan.

**Comment by kimrutherford** [comment]
> I think adding them to Chado would be a good plan.

I've made a separate issue about protein_ids: pombase/pombase-chado#1090

**Comment by kimrutherford** [comment]
From ENA:

> Regarding your submission, please go ahead as it should work. It should go
> through as long as the locus tag after 'SPOM_' is not within what is shown
> below. For example, SPBC1348.15 should go through.
>
> "text"(single token) but not "<1-5 letters><5-9 digit integer>[.<integer>]"

**Comment by kimrutherford** [comment]
Add the pombe systematic ID as a `/db_xref="PomBase:SP..."`

Use `/locus_tag="SPOM_SP..."`

**Comment by kimrutherford** [comment]
> Add the pombe systematic ID as a /db_xref="PomBase:SP..."
> Use /locus_tag="SPOM_SP..."

With that change, there are still a bunch of errors but there are no complaints about /locus_tag

Here are the remaining errors: 

```
ERROR: "5'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 13771 of chromosome1.embl.gz,  line: 13764 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 932 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 7822 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 16755-16756 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 19294-19295 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 28031 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 30088-30089 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 31117-31118 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 31117-31118 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 37676 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 38545 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 41435-41436 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 41509-41512 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 46062 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 48939 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 52614-52615 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 52642 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 61728 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 61728 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 61765 of chromosome1.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 1656 of chromosome2.embl.gz,  line: 1652 of chromosome2.embl.gz]
ERROR: "5'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 1696 of chromosome2.embl.gz,  line: 1692 of chromosome2.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 8025 of chromosome2.embl.gz,  line: 8021 of chromosome2.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 16399 of chromosome2.embl.gz,  line: 16395 of chromosome2.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 44746 of chromosome2.embl.gz,  line: 44742 of chromosome2.embl.gz]
ERROR: Mandatory qualifier "ncRNA_class" not present in feature "ncRNA". [ line: 49170 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 8744 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 8975-8976 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 11509 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 22759-22760 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 22872 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 22928 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 23156-23157 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 23156-23157 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 23931 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 23984 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 24718 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 26127 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 32140 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 33293 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 36296 of chromosome2.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 43469-43471 of chromosome2.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 2304 of chromosome3.embl.gz,  line: 2300 of chromosome3.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 17567 of chromosome3.embl.gz,  line: 17563 of chromosome3.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 22483 of chromosome3.embl.gz,  line: 22479 of chromosome3.embl.gz]
ERROR: "3'UTR" Features locations are duplicated - consider merging qualifiers. [ line: 24062 of chromosome3.embl.gz,  line: 24058 of chromosome3.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 19766 of chromosome3.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 25970 of chromosome3.embl.gz]
```

**Comment by ValWood** [comment]
Not too bad!

Can you give me some of the example intron lines. These could be the frameshifts (but we represent these as single base overlaps - as requested by Ensembl) , so I would be surprised if they are recognised as an intron.

I first want to check that they aren't pseudos.

**Comment by ValWood** [comment]
Also If you let me know which the duplicated UTRs are I can fix them....probably

**Comment by ValWood** [comment]
This is an odd error:

ERROR: Mandatory qualifier "ncRNA_class" not present in feature "ncRNA". [ line: 49170 of chromosome2.embl.gz]

I have never heard of ncRNA_class,  which feature is this?

**Comment by kimrutherford** [comment]
It's this one which I think is our only ncRNA feature:
```
FT   ncRNA           4071111..4071312
FT                   /db_xref="PMID:29914874"
FT                   /db_xref="EMBL:AU013717"
FT                   /product="small non-coding RNA"
FT                   /systematic_id="SPNCRNA.105"
FT                   /controlled_curation="term=warning, no detectable long
FT                   open reading frame; qualifier=predicted;
FT                   db_xref=EMBL:AU013717; date=20050412"
```

All the other RNA have a more specific type like tRNA, lncRNA or snoRNA

The code that generates the EMBL format files for submission changes the feature type to ncRNA and puts the RNA type in `/ncRNA_class' like this:

```
FT   ncRNA           complement(2347..2688)
FT                   /product="non-coding RNA"
FT                   /ncRNA_class="lncRNA"
FT                   /locus_tag="SPOM_SPNCRNA.4512"
```

https://www.ebi.ac.uk/ena/WebFeat/qualifiers/ncRNA_class.html

Is there a more specific type we can give to SPNCRNA.105?  We can use "other" as the type if needed: https://www.insdc.org/submitting-standards/ncrna-vocabulary/

**Comment by ValWood** [comment]
weird, I thought they were all typed
https://www.pombase.org/results/from/id/25f2ca85-35d8-4faa-86a1-8ab259150e01
will take a look

**Comment by ValWood** [comment]
This was 202 residues so using the arbitrary cut-off, it is officially a lncRNA, fixed

**Comment by kimrutherford** [comment]
> weird, I thought they were all typed

"ncRNA" works a bit weirdly in the query builder because of this:
https://github.com/pombase/website/issues/1981#issuecomment-1384678405

If you select "ncRNA" it's configured to query for any of: rRNA tRNA snRNA snoRNA lncRNA sncRNA
The list doesn't include "ncRNA" because we decided to get rid of ncRNA features.

**Comment by kimrutherford** [comment]
> Also If you let me know which the duplicated UTRs are I can fix them....probably

I'm wondering if we will need to adjust how we represent transcripts for this submission.  I'll see how features look for other MODs on Monday.

**Comment by ValWood** [comment]
I knew there was some weirdness, that one must have slipped through

**Comment by kimrutherford** [comment]
> Can you give me some of the example intron lines. 

This should be all of them:  [embl_warnings.txt](https://github.com/pombase/curation/files/13186979/embl_warnings.txt)

**Comment by kimrutherford** [comment]
> I'm wondering if we will need to adjust how we represent transcripts for this submission. I'll see how features look for other MODs on Monday.

I've just spent some trying to find documentation or examples showing how UTRs should be representation in an EMBL file.  I didn't find any useful documentation.

I found this example of a C. elegans gene where there are no explicit UTRs, only mRNA and CDS features:
https://www.ncbi.nlm.nih.gov/nuccore/NC_003281.10?from=8975069&to=8981897&report=genbank&strand=true

Maybe that's how we should be doing it for our submission?

**Comment by kimrutherford** [comment]
Here's a Drosophila example:
https://www.ncbi.nlm.nih.gov/nuccore/NT_033778.4?report=genbank&from=21323430&to=21341847

**Comment by kimrutherford** [comment]
- Remove SPD db_xrefs
 - change `/standard_name` to /gene
 - pass through `/ribosomal_slippage`
 - add `gene` features
 - add `mRNA` features

**Comment by ValWood** [comment]
Remove SPD db_xrefs
was the thing I couldn't remember

**Comment by kimrutherford** [comment]
While removing the SPD db_xrefs I noticed a few other odd xrefs.  Here are the counts and prefix:

```
      1 GenBank
      1 SO
      4 Rfam
    113 KEGG
    864 EMBL
```

Should I remove those too?

**Comment by kimrutherford** [comment]
> add gene features
> add mRNA features

The new plan is to rely on the UTR features.  There is currently a warning about a duplicate UTR.  We'll fix that by dropping on of the features.

**Comment by kimrutherford** [comment]
> change /standard_name to /gene

We get different errors after that change:

```
ERROR: Features sharing locus_tag "SPOM_SPAC144.19" are associated with "gene" qualifiers with different values ("SPAC144.19" and "prl46"). [ line: 53775 of chromosome1.embl.gz]
ERROR: Features sharing locus_tag "SPOM_SPCC1919.01" are associated with "gene" qualifiers with different values ("ckk2" and "ppk34"). [ line: 23884 of chromosome3.embl.gz]
```

In the first case the 3'UTR has `/gene="SPAC144.19"` and the CDS has `/gene="prl46"`.
In the second the 3'UTR has `/gene="ppk34"` and the CDS has `/gene="ckk2"`

These don't case problems for PomBase loading because we ignore /gene on UTRs.

**Comment by kimrutherford** [comment]
> The new plan is to rely on the UTR features. There is currently a warning about a duplicate UTRs. We'll fix that by dropping one of the features.

That worked.  There are now no warnings about duplicate UTRs.

Oddly, it doesn't warn about the two genes that have duplicate CDSs (mei3 and pgr1)

I've also fixed the code to use the gene systematic ID in the /locus_tag for multi transcript genes.  It was getting that wrong and using the transcript ID.

The submission files have /protein_id qualifiers and UniProtKB/Swiss-Prot dbxrefs for the CDS features.  We currently do that even when there are two transcripts.  I wonder if that makes sense since each transcript will probably produce a different protein.  Do you think that's OK?

**Comment by kimrutherford** [comment]
I think the only issues now are the [warnings about `/gene`](https://github.com/pombase/curation/issues/3345#issuecomment-1786064957) and the frameshifts

**Comment by ValWood** [comment]
>While removing the SPD db_xrefs I noticed a few other odd xrefs. Here are the counts and prefix:
>Should I remove those too?

Yes please, we don't import, use or maintain  them

**Comment by ValWood** [comment]
Re:
>In the first case the 3'UTR has /gene="SPAC144.19" and the CDS has /gene="prl46".
>In the second the 3'UTR has /gene="ppk34" and the CDS has /gene="ckk2"
>These don't case problems for PomBase loading because we ignore /gene on UTRs.

I don't maintain gene names on UTRs, I only add systematic IDs (otherwise I'd need to add the names in 3 places)
All UTRS should have a systematic ID, and we can delete /gene from UTRs in the contigs

**Comment by ValWood** [comment]
>The submission files have /protein_id qualifiers and UniProtKB/Swiss-Prot dbxrefs for the CDS features. We currently do >that even when there are two transcripts. I wonder if that makes sense since each transcript will probably produce a >different protein. Do you think that's OK?

I think the PID should be on the .1 transcript only.

**Comment by kimrutherford** [comment]
> I don't maintain gene names on UTRs, I only add systematic IDs (otherwise I'd need to add the names in 3 places)
> All UTRS should have a systematic ID, and we can delete /gene from UTRs in the contigs

OK, thanks.  I've removed /gene from the UTRs and we're now left with the 37 frame-shift errors:

```
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 831 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 6568 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 13977-13978 of chromosome1.embl.gz]
ERROR: Intron usually expected to be at least 10 nt long. Please check the accuracy. [ line: 16079-16080 of chromosome1.embl.gz]
...
```

**Comment by kimrutherford** [comment]
> I think the PID should be on the .1 transcript only.

OK, thanks.  I've made that change and it didn't cause any extra errors.

**Comment by kimrutherford** [comment]
>> While removing the SPD db_xrefs I noticed a few other odd xrefs. Here are the counts and prefix:
>> Should I remove those too?
>
> Yes please, we don't import, use or maintain  them

That's done as well.

**Comment by ValWood** [comment]
Thanks, so waiting for ENA and then we are good to know. I'll let Uniprot know we are  (hopefully) close!

**Comment by ValWood** [comment]
I'll close this ticket as we aren't using it !
